src/keys/slip10.rs

```diff
@@ -33,16 +30,20 @@ mod hazmat {
     pub trait IsPublicKey: Derivable {
         type SecretKey: IsSecretKey;
     }
+    pub trait ToPublic: IsSecretKey {
+        fn to_public(sk_bytes: &[u8; 33]) -> [u8; 33];
+    }
+    pub trait CalcData<S: Segment> {
+        fn calc_data(key_bytes: &[u8; 33], segment: S) -> [u8; 33];
```

Comment:
What's `calc_data` doing?  Calculating the public key or what data?
The name isn't really helpful

Reply:
Agree. It calculates part of 'data' argument to be hashed in hmac which is either a secret key or a public key. It's implementation detail and it has to be `pub`.